### PR TITLE
Import Atmel SAML21, SAMR34 and SAMR35 header files from ASF library

### PR DESCRIPTION
**Pull request by attie-argentum**
Zephyr support for these parts is following, but depends on this PR.

This PR supersedes #11, which can likely be closed now.

### CODE REVIEW

**Comment by attie-argentum** [comment]
@stephanosio / @nandojve / @mnkp - any chance we can look at getting this merged?

**Comment by attie-argentum** [comment]
@stephanosio / @nandojve / @mnkp - Hi all, what's the typical process here? Does zephyrproject-rtos/zephyr#41319 need to be approved before this will be considered, or can we get this merged and on the way?

**Comment by nandojve** [comment]
Hi @attie-argentum ,

The hal code is merged when people is reviewing 41308. Someone need check why scancode is not happy. Did you look at it?
Right now, Zephyr is in stabilization phase due to new v3.0.0 release, see [release process](https://docs.zephyrproject.org/latest/development_process/release_process.html). This means that only documentation and bug fixes will be accepted. We need wait next merge window to move forward with new SoC and/or features.

**Comment by attie-argentum** [comment]
Hi @nandojve ,

Understood, thanks very much.

Re scancode: This appears to be a license issue - other existing files will likely fail this check too (e.g: [samd21/component-version.h](https://github.com/zephyrproject-rtos/hal_atmel/blob/master/asf/sam0/include/samd21/component-version.h) is also `apache-2.0`)... I'm not comfortable forcibly relicensing these files to `bsd-x11` just to get the check to pass...

**Review by stephanosio — APPROVED** [review]
This PR, in itself, looks ok.

> Re scancode: This appears to be a license issue - other existing files will likely fail this check too (e.g: [samd21/component-version.h](https://github.com/zephyrproject-rtos/hal_atmel/blob/master/asf/sam0/include/samd21/component-version.h) is also apache-2.0)... I'm not comfortable forcibly relicensing these files to bsd-x11 just to get the check to pass, and this doesn't appear to have been done in the past...

The failure does not have to do with the `apache-2.0` licensed ones, but rather with the ones that supposedly have "non-permissive" licenses:

```
...
* scan/asf/sam0/include/saml21/saml21g17b.h has non-permissive license: nokos-1.0a
...
* scan/asf/sam0/include/saml21/component/ac.h has non-permissive license: elastic-license-2018
...
```

Since it looks like those are mis-detections and all files are licensed Apache 2.0, the scancode failure can be ignored.

**Review by nandojve — APPROVED** [review]

**Comment by attie-argentum** [comment]
> rather with the ones that supposedly have "non-permissive" licenses

Ah, understood - thanks!

**Comment by attie-argentum** [comment]
Rebase

**Comment by nandojve** [comment]
Hi @attie-argentum ,

Please, rebase your code and:

1- check this commit 45cb8f898ffe6ee932fb6d983a113d4e280be40b and make sure you version not suffer same problem
2- you need add pinconfig files + headers because now all drivers are using pinctrl API. You can look at pinconfig [README](https://github.com/zephyrproject-rtos/hal_atmel/blob/2c2ce9f449a3e61b56b0de624007a81a77037f4d/pinconfigs/README.md)  for guidance.

**Comment by attie-argentum** [comment]
Thanks @nandojve - I'm hoping to work on this soon, and I'll check your points.

Very well done with zephyrproject-rtos/zephyr#43731 - I'm planning to follow suit with both this and zephyrproject-rtos/zephyr#41308.

**Comment by attie-argentum** [comment]
Points 1 & 2 are already resolved... looking in to 3 now.

File: pinconfigs/sam-l21-ab.yml [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+
+family: l21
+
+map: SAM_PINMUX
+
+series: [l21]
+
+revisions:
+  l21: ab
+
```
Comment (nandojve) [inline pinconfigs/sam-l21-ab.yml:29]: No need to add revisions for this variant.

File: pinconfigs/sam-r34-r35-b.yml [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+family: r34-r35
+
+map: SAM_PINMUX
+
+series: [r34, r35]
+
+revisions:
+  r34: b
+  r35: b
+
```
Comment (nandojve) [inline pinconfigs/sam-r34-r35-b.yml:27]: No need to add revisions for this variant.

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @attie-argentum,

Please, drop revisions to keep pinconfig generic.

File: pinconfigs/sam-l21-ab.yml (hunk 1 above)
Comment (attie-argentum) [reply pinconfigs/sam-l21-ab.yml]: done

File: pinconfigs/sam-r34-r35-b.yml (hunk 2 above)
Comment (attie-argentum) [reply pinconfigs/sam-r34-r35-b.yml]: done

**Comment by attie-argentum** [comment]
@nandojve thanks - done

**Review by nandojve — CHANGES_REQUESTED** [review]
Hi @attie-argentum ,

I forget to mention but since the `yml` file is generic, the revision should not be part of file name. Please, drop revision from both *.yml file names.

I'm ready to approve as soon you send update.

**Comment by attie-argentum** [comment]
Ah yes, my apologies! Done.

**Review by stephanosio — APPROVED** [review]
Looks reasonable to me.

**Pull request merged**